wordpress-mobile/WordPress-Android
# Remove browser type from media picker

This PR removes the browser type from the MediaPickerActivity. There are 2 tricky places where it was still used and I hope I fixed them correctly.

I've moved this part:
```
if (browserType == FEATURED_IMAGE_PICKER) {
                    featuredImageHelper.trackFeaturedImageEvent(
                            IMAGE_PICKED,
                            localPostId
                    )
                }
```
to the place where the results are processed in the `EditPostSettingsFragment`. I've tested it and I think it's tracking the upload correctly.

I've removed the `isWPStoriesMediaBrowserTypeResult(data))` from the stories handler. As far as I can tell it wasn't necessary. I've checked all the usage and I don't think there is any branch we're skipping or missing if the check failed (I don't think it could've failed, it was just a precaution). Let me know if it makes any sense. 

I've squeezed in `LiveDataUtils` update into this PR because it will be necessary for the stock media data source. I hope that's fine. 

To test:
- Set the ConsolidatedMediaPickerConfig to `true` in the App settings
- Check the featured image upload
- Check that an event like this `WordPress-STATS: 🔵 Tracked: featured_image_picked_post_settings, Properties: {"post_id":2}` is tracked correctly when the image is picked (before the upload finishes)
- Check that you can pick media for stories

PR submission checklist:
